alf/data_structures.py

```diff
@@ -110,6 +111,39 @@ def is_last(self):
         raise ValueError('step_type is not a Torch Tensor')
 
 
+def _create_timestep(observation, prev_action, reward, discount, env_id,
+                     step_type):
+    discount = torch.tensor(discount)
+    # Warning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
+    # Input struct might contain tensor.
+    # TODO: remove warning
+    create_tensor = lambda t: t.clone().detach() if torch.is_tensor(
```

Comment:
Since there is no gradient required for any of the fields, I guess we don't have to do clone. Also, all these inputs should already be tensors (they are returned by the env @runjerry ).

Reply:
just using detach().
tf_agents has this, so just copying over in case needed.